python/ray/_private/runtime_env/nsight.py

```diff
@@ -0,0 +1,132 @@
+import os
+import logging
+import subprocess
+import copy
+from typing import Tuple, List, Dict, Optional
+
+from ray._private.runtime_env.context import RuntimeEnvContext
+from ray._private.runtime_env.plugin import RuntimeEnvPlugin
+from ray._private.utils import (
+    try_to_create_directory,
+)
+
+default_logger = logging.getLogger(__name__)
+
+NSIGHT_DEFAULT_CONFIG = {
+    "-t": "cuda,cudnn,cublas,nvtx",
+    "-o": "'worker_process_%p'",
+    "--cudabacktrace": "all",
+    "--stop-on-exit": "true",
+}
+
+
+def check_nsight_script(nsight_config: Dict[str, str]) -> Tuple[bool, str]:
```

Comment:
can you add unit test for this method?

Reply:
For this method, unit test is already covered by `test_check_invalid_option`, so adding one for this might be redundant

Reply:
Hmm is it really true? For example, the else (return_code != 0), is this covered? We should make sure to test 

1. return code == 0 -> file is deleted
2. return code != 0 -> file is deleted